### ISSUE TITLE
Erubi Support

### ISSUE THREAD

**Issue by adam12** [padrino/padrino-framework#2089]
I think it would be worth considering support for [Erubi](https://github.com/jeremyevans/erubi), either in addition or in replacement of Erubis. I won't go into the benefits of Erubi - Jeremy Evans does a great job in the README.

I ran some benchmarks using my code here: https://gist.github.com/adam12/8c3613d0c9ba6ebddc8eb8a962b36323

In summary:

* Erubis vs Erubi performance fairly comparable.
* Memory usage of Erubi is indeed 90% lower (per claim in Erubi README).
* Jeremy is known to be an excellent OSS maintainer.

(the following tests run with `ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]`)
### Performance Benchmark
```
Comparison:
1k fast erubis    (eval):        3.4 i/s
1k erubis         (eval):        3.2 i/s - 1.04x  slower
1k erb            (eval):        3.1 i/s - 1.08x  slower
1k erubi          (eval):        2.7 i/s - 1.22x  slower
1k erubi  (compile+eval):        2.1 i/s - 1.58x  slower
1k erubis (compile+eval):        2.1 i/s - 1.61x  slower
1k erb    (compile+eval):        1.9 i/s - 1.76x  slower
```

### Memory Benchmark
*Some data removed for brevity*

```
allocated memory by gem
-----------------------------------
    102227  erubis-2.7.0
     15992  erubi-1.0.0

allocated objects by gem
-----------------------------------
       908  erubis-2.7.0
       181  erubi-1.0.0

retained memory by gem
-----------------------------------
     50999  erubis-2.7.0
      2528  erubi-1.0.0

retained objects by gem
-----------------------------------
       154  erubis-2.7.0
         8  erubi-1.0.0
```

I'd be willing to submit a PR should this be of interest.

**Comment by ujifgc** [comment]
Yes, you are welcome to submit a PR.

**Comment by wikimatze** [comment]
@adam12 do you want to implement the support for this?

**Comment by adam12** [comment]
@wikimatze I'm going to explore adding it, yes.

**Comment by wikimatze** [comment]
Perfect, looking forward to your contribution.

**Comment by fidalgo** [comment]
Actually Rails is moving to Erubi also: https://github.com/rails/rails/pull/27757

**Comment by ujifgc** [comment]
Implemented in #2090
